elliptic-curve/src/group_digest.rs

```diff
@@ -0,0 +1,86 @@
+use crate::hash2field::{hash_to_field, ExpandMsg, FromOkm};
+use crate::isogeny::Isogeny;
+use crate::osswu::OsswuMap;
+use ff::Field;
+use group::Group;
+
+/// Convert a set of field elements into a point
+pub trait ConvertToPoint {
+    /// The field element that represents a point
+    type FieldElement: Field;
+    /// The result of converting two field elements to a Point
+    type Output;
+
+    /// Converts two coordinates into a point
+    fn convert_to_point(x: Self::FieldElement, y: Self::FieldElement) -> Self::Output;
+}
+
+/// Method for points to be cleared by the curve cofactor
+pub trait ClearCofactor {
+    /// Multiply this point by the curve cofactor to ensure
+    /// correct subgroup membership
+    fn clear_cofactor(&mut self);
+}
+
+/// Adds hashing arbitrary byte sequences to a valid group element
+pub trait GroupDigest {
+    /// The field element representation for a group value with multiple elements
+    type FieldElement: FromOkm + OsswuMap + Isogeny;
+    /// The resulting group element
+    type Output: Group
+        + ClearCofactor
+        + ConvertToPoint<FieldElement = Self::FieldElement, Output = Self::Output>;
```

Comment:
I think this was done to avoid additional `where` bounds because associated type bounds isn't stable?
As pointed above I think a missing trait here would be `MapToCurve` that would fill the gap without requiring additional bounds on the `FieldElement`.